planetiler-core/src/test/java/com/onthegomap/planetiler/reader/GeoPackageReaderTest.java

```diff
@@ -66,13 +68,44 @@ void testReadGeoPackage(boolean keepUnzipped) throws IOException {
     }
   }
 
+
+  @ParameterizedTest
+  @CsvSource({
+    "geopackage.gpkg,4",
+    "geopackage-unindexed.gpkg,86"
+  })
+  @Timeout(30)
+  void testReadGeoPackageSpatialIndex(String dbName, int expectedCount) throws IOException {
+    Path path = TestUtils.pathToResource(dbName);
+
+    var proj =  "EPSG:4326";
+
+    try (
+      var reader =
+        new GeoPackageReader(proj, "test", path, tmpDir, false, new Envelope(-77.0306, -77.0192, 38.8894, 38.9014))
+    ) {
+      for (int iter = 0; iter < 2; iter++) {
+        String id = "path=" + path + " proj=" + proj + " iter=" + iter;
+        assertEquals(86, reader.getFeatureCount(), id);
+        List<Geometry> points = new ArrayList<>();
+        WorkerPipeline.start("test", Stats.inMemory())
+          .fromGenerator("geopackage", reader::readFeatures, 1)
+          .addBuffer("reader_queue", 100, 1)
+          .sinkToConsumer("counter", 1, elem -> {
+            points.add(elem.latLonGeometry());
+          }).await();
+        assertEquals(expectedCount, points.size(), id);
```

Comment:
Do we need WorkerPipeline here? I think we could potentially simplify this to just:

```java
reader.readFeatures(feature -> {
  points.add(elem.latLonGeometry());
});
```